ambry-network/src/main/java/com/github/ambry/network/BlockingChannelInfo.java

```diff
@@ -169,6 +169,8 @@ private BlockingChannel getBlockingChannelBasedOnPortType(String host, int port)
       channel = new BlockingChannel(host, port, config);
     } else if (this.port.getPortType() == PortType.SSL) {
       channel = new SSLBlockingChannel(host, port, registry, config, sslSocketFactory, sslConfig);
+    } else {
+      logger.error("No port type: {}", this.port.getPortType());
```

Comment:
Returning null here in this case seems a bit weird to me instead of throwing an exception, or throwing an exception in the constructor if it is misconfigured. Is there a reason for that behavior? I was also wondering why this private method takes in the port number instead of using `port.getPort()` (I realize this may be very old code).

Reply:
Updated the code to throw an exception. (BTW, The piece of  code will be deprecated after we no longer need blocking channel.)